### DISCUSSION TITLE
[V5] How to send POST request to Google reCaptcha?

### DISCUSSION THREAD

**Discussion by HappyLDE** [adonisjs/core#2331] [Help] [answered]
Hi,

I need to make a request to the Google reCaptcha API by sending a POST request here :
https://www.google.com/recaptcha/api/siteverify

And I need to include the following:
```
{
  "secret": "the-secret-key",
  "response": "client-side-response",
  "remoteip": "client-ip",
  "action": "contact",
  "threshold": "0.5",
  "hostname": "host-where-the-captcha-was-solved",
}
```

How can I do this in adonisjs V5 (the request)? Thank you!

**Comment by max-bertinetti** [comment, marked as answer]
Hi @HappyLDE.
Some times ago there was an [Adonis Recaptcha plugin](https://github.com/masasron/adonis-recaptcha).
It was using request that now is deprecated.
I think someone can revamp this for V5 someday.
For now I can suggest you to try [node-fetch](https://github.com/node-fetch/node-fetch)

I don't know if @thetutlage or @RomainLanz agree with me.

**Reply by HappyLDE** [reply]
Thank you @maxbertinetti I tried node-fetch but had some problems with the await giving errors so I ended up using this solution instead:
https://nodejs.dev/learn/making-http-requests-with-nodejs

Edit: I'd like to mention that Google reCaptcha requires the parameters as string in the link and not as array parameters.

Edit2: Turns out the await problem came because I used it in a forEach loop instead of a for of, so node-fetch should be good too. Setting your answer as the accepted answer.

**Comment by RomainLanz** [comment]
Hey @HappyLDE! 👋 

An easy solution is to use [got](https://github.com/sindresorhus/got).
